pkg/api/apikey.go

```diff
@@ -82,12 +81,9 @@ func (hs *HTTPServer) AddAPIKey(c *models.ReqContext) response.Response {
 			return response.Error(400, "Number of seconds before expiration is greater than the global limit", nil)
 		}
 	}
+
+	cmd.ServiceAccountId = nil // API keys can't be added to SAs through this endpoint
```

Comment:
should we have a message or something coming to the user, if there is a serviceaccountID, in the request?

Reply:
This parameter is not exposed in the documentation for this endpoint, so there should be no expectation of the user using it